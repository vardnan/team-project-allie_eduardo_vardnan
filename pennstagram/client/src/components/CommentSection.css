.commentSection {
    width: 100%;
    height: 100%;
}

.userComment {
    display: flex;
    flex-direction: row;
    padding: 10px;
    align-items: center;
    margin: 0;

}

.userComment img {
    order: 0;
    border-radius: 100%;
    width: 40px;
    height: 40px;
    padding: 0;
    margin: 0;
    margin-right: 10px;
}

.userComment h3 {
    order: 1;
    margin: 0;
    height: 100%;
    width: 100%;
    border-bottom: 1px;
    border-color: black;
    font-size: 11pt;
}

#commentText {
    display: flex;
    flex-direction: column;
    width: 100%;
}

#commentText p {
    order: 2;
    margin: 0;
    font-size: 11pt;
    padding-top: 5px;
    height: 100%;
    width: 100%;
}


.commentedUser {
    display: flex;
    flex-direction: row;
    align-items: center;
    height: 100%;
    width: 100%;
}

.caption {
    display: flex;
    flex-direction: row;
    padding: 10px;
    align-items: center;
    margin: 0;

}

.caption img {
    order: 0;
    border-radius: 100%;
    width: 40px;
    height: 40px;
    padding: 0;
    margin: 0;
    margin-right: 10px;
}

.caption h3 {
    order: 1;
    margin: 0;
    height: 100%;
    width: 100%;
    border-bottom: 1px;
    border-color: black;
    font-size: 11pt;
} 

.caption p {
    margin: 0;
    font-size: 11pt;
}

.postUser {
    display: flex;
    flex-direction: row;
    align-items: center;
}
#captionText {
    display: flex;
    flex-direction: column;
    width: 100%;
}

#captionText h3 {
    order: 0;
}

#captionTest p {
    order: 2;
    margin: 0;
    font-size: 11pt;
    padding-top: 5px;
    height: 100%;
    width: 100%;
}
