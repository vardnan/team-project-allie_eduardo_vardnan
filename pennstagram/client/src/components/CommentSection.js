import React from 'react';
import PropTypes from 'prop-types';
import './CommentSection.css';

function CommentSection({
  comments, caption, profileImage, postUser,
}) {
  return (
    <div>
      <div className="commentSection">
        <div className="caption">
          <div className="postUser">
            <img src={profileImage} alt={`${postUser}'s profile`} />
            <div id="captionText">
              <h3>{postUser}</h3>
              <p>{caption}</p>
            </div>
          </div>
          <div />
        </div>
        {comments.map((comment) => {
          const profileImageUrl = comment.commenterProfileImage
            ? comment.commenterProfileImage
            : 'https://picsum.photos/200';
          return (
            <div key={comment.id}>
              {' '}
              <div className="userComment">
                <div className="commentedUser">
                  <img
                    src={profileImageUrl}
                    alt={`${comment.username}'s profile`}
                  />
                  <div id="commentText">
                    <h3>{comment.username}</h3>
                    <p>{comment.comment}</p>
                  </div>
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
}

CommentSection.propTypes = {
  comments: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      username: PropTypes.string.isRequired,
      comment: PropTypes.string.isRequired,
    }),
  ).isRequired,
  caption: PropTypes.string.isRequired,
  profileImage: PropTypes.string.isRequired,
  postUser: PropTypes.string.isRequired,
};

export default CommentSection;
