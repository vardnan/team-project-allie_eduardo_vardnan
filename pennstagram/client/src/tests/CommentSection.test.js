import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CommentSection from '../components/CommentSection';
import UserContext from '../contexts/user-context';

// Mock Data
const mockUser = {
  username: 'Alice',
  profileImage: 'alice.jpg',
};

const mockAllUsers = [
  { username: 'Alice', profileImage: 'alice.jpg' },
  { username: 'Bob', profileImage: 'bob.jpg' },
];

const mockComments = [
  {
    username: 'Bob',
    comment: 'Great post!',
  },
  {
    username: 'Alice',
    comment: 'Thank you!',
  },
];

// Mock Context
const mockContext = {
  allUsers: mockAllUsers,
};

// Testing rendering
test('Testing that the CommentSection is rendering correctly', () => {
  const props = {
    comments: mockComments,
    caption: 'Caption text',
    profileImage: mockUser.profileImage,
    postUser: mockUser.username,
  };

  const { getByText } = render(
    <UserContext.Provider value={mockContext}>
      <CommentSection {...props} />
    </UserContext.Provider>
  );
  expect(getByText('Caption text')).toBeInTheDocument();
  expect(getByText('Great post!')).toBeInTheDocument();
  expect(getByText('Thank you!')).toBeInTheDocument();
});

// Snapshot testing
test('Testing that the CommentSection matches snapshot', () => {
  const props = {
    comments: mockComments,
    caption: 'Caption text',
    profileImage: mockUser.profileImage,
    postUser: mockUser.username,
  };

  const component = renderer.create(
    <UserContext.Provider value={mockContext}>
      <CommentSection {...props} />
    </UserContext.Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
