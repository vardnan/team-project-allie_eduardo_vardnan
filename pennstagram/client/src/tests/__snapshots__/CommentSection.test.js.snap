// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing that the CommentSection matches snapshot 1`] = `
<div>
  <div
    className="commentSection"
  >
    <div
      className="caption"
    >
      <div
        className="postUser"
      >
        <img
          src="alice.jpg"
        />
        <div
          id="captionText"
        >
          <h3>
            Alice
          </h3>
          <p>
            Caption text
          </p>
        </div>
      </div>
      <div />
    </div>
    <div>
      <div
        className="userComment"
      >
        <div
          className="commentedUser"
        >
          <img
            alt="Bob's profile"
            src="bob.jpg"
          />
          <div
            id="commentText"
          >
            <h3>
              Bob
            </h3>
            <p>
              Great post!
            </p>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="userComment"
      >
        <div
          className="commentedUser"
        >
          <img
            alt="Alice's profile"
            src="alice.jpg"
          />
          <div
            id="commentText"
          >
            <h3>
              Alice
            </h3>
            <p>
              Thank you!
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
